Bali/src/main/java/OfferList.java

```diff
@@ -0,0 +1,117 @@
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Locale;
+import java.util.stream.Collectors;
+
+public class OfferList {
+    private List<Offer> offers;
+
+    public OfferList(){
+        offers = new ArrayList<>();
+    }
+
+    public Offer getOffer(String id){
+        for(Offer offer : offers){
+            if(offer.getId().equals(id)){
+                return offer;
+            }
+        }
+        System.out.println("Incorrect type of id!");
```

Comment:
What do you mean by "incorrect type of id"?) Type is String, this is said in the method params))
And please avoid writing to stdout in any client-server app. This output goes to literally nowhere, and no one will see it (servers are usually unattended, "have no display"). The only way is to write to a file. For that, you need a logger (google for SLF4J)